### PR TITLE
refactor: add hub as parameter

### PR DESCRIPTION
**Pull request by 0xsambugs**
We are now taking the `DCAHub` and the `DCAPermissionManager` as parameters on the Companion. Also, we changed the docs so that they were consistent with are new repositories

### CODE REVIEW

File: test/unit/DCAHubCompanion/dca-hub-companion-multi-call-handler.spec.ts [hunk 1]
```diff
@@ -247,6 +278,7 @@ contract('DCAHubCompanionMulticallHandler', () => {
     params: [10, constants.NOT_ZERO_ADDRESS, constants.NOT_ZERO_ADDRESS],
   });
 
+  type DropFirstParam<T extends unknown[]> = T extends [any, ...infer U] ? U : never;
```
Comment (0xged) [inline test/unit/DCAHubCompanion/dca-hub-companion-multi-call-handler.spec.ts:281]: Some voodoo sheit here.

**Review by 0xged — APPROVED** [review]
Looks good! This is becoming interesting.

File: test/unit/DCAHubCompanion/dca-hub-companion-multi-call-handler.spec.ts (hunk 1 above)
Comment (0xsambugs) [reply test/unit/DCAHubCompanion/dca-hub-companion-multi-call-handler.spec.ts:281]: I googled this haha

**Review by nchamo — APPROVED** [review]
Thanks for adding consistency to the docs 😊

**Pull request merged**